### ISSUE TITLE
Add back `cid` option in `block.put` upstream (js-ipfs)

### ISSUE THREAD

**Issue by oed** [ceramicnetwork/js-ceramic#1881]
~~Make  [`storeCommit`](https://github.com/ceramicnetwork/js-ceramic/blob/76341747a4bc9aee1593e8663b94413a07748d94/packages/core/src/dispatcher.ts#L84) take an optional parameter `blockCID` that when given uses `block.put` instead of `dag.put`~~
```ts
await ipfs.block.put(data, { cid: blockCID })
```

~~Make [`createStreamFromGenesis`](https://github.com/ceramicnetwork/js-ceramic/blob/76341747a4bc9aee1593e8663b94413a07748d94/packages/core/src/ceramic.ts#L566) take an optional parameter `blockCID` that get's passed to `storeCommit`.~~

~~Make [`TileDocument.createFromGenesis`](https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/stream-tile/src/tile-document.ts#L202) take an optional parameter `blockCID` that get's passed to `createStreamFromGenesis`~~

---

Above is no longer needed since the issue for us is in a [`linkedBlock`](https://github.com/ceramicstudio/datamodels/blob/main/packages/identity-profile-basic/src/model.ts#L56) and we already use [`block.put`](https://github.com/ceramicnetwork/js-ceramic/blob/76341747a4bc9aee1593e8663b94413a07748d94/packages/core/src/dispatcher.ts#L91) for that.

However, there is a bug https://github.com/ipfs/js-ipfs/issues/3960 in js-ipfs that disallows passing of the CID to `block.put`. This ticket now tracks adding that option back upstream.

**Comment by ukstv** [comment]
Relevant issue = blocker: https://github.com/ipfs/js-ipfs/issues/3960

**Comment by stbrody** [comment]
why would we need to provide the CID?  Isn't the CID derived from the block contents?

**Comment by ukstv** [comment]
Well, true. My mistake, I guess. Block CID is recalculated based on passed codec name and hash name only. No decoding is involved. So, we do not have to pass a CID there. It is calculated all right.

**Comment by oed** [comment]
Well if you have the CID you might as well include it, otherwise you need to get the multihash type and the codec name and pass those as options.
@ukstv why was this closed? I changed it for us to track the upstream PR to add this functionality back.

**Comment by ukstv** [comment]
@oed We are not blocked by the upstream feature, that's why. Passing block.put with proper name of the codec (which we already know based on CID), achieves the same result as passing a cid. For sure, this only works for a codec that is already included in IPFS, like `dag-cbor`.

**Comment by oed** [comment]
Ok, that works. Do we have a new ticket to do it in that way?

Sent via Superhuman iOS ***@***.***>

**Comment by ukstv** [comment]
I am busted. Now there is: https://github.com/ceramicnetwork/js-ceramic/issues/1894

**Comment by oed** [comment]
Thanks!

Sent via Superhuman iOS ***@***.***>
